src/Riok.Mapperly/Helpers/EnumNamingStrategyHelper.cs

```diff
@@ -0,0 +1,59 @@
+using System.Globalization;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis;
+using Riok.Mapperly.Abstractions;
+
+namespace Riok.Mapperly.Helpers;
+
+public static class EnumNamingStrategyHelper
+{
+    public static Dictionary<IFieldSymbol, string> BuildCustomNameStrategyMappings(
+        this ITypeSymbol enumSymbol,
+        EnumNamingStrategy namingStrategy
+    )
+    {
+        var customNameMappings = new Dictionary<IFieldSymbol, string>(SymbolEqualityComparer.Default);
+
+        var values = enumSymbol.GetMembers().OfType<IFieldSymbol>();
+        foreach (var value in values)
+        {
+            var valueString = ConvertEnumValueNameToString(value.Name, namingStrategy);
+            customNameMappings.Add(value, valueString);
+        }
+
+        return customNameMappings;
+    }
+
+    private static string ConvertEnumValueNameToString(string enumValueName, EnumNamingStrategy namingStrategy)
+    {
+        return namingStrategy switch
+        {
+            EnumNamingStrategy.PascalCase => enumValueName,
+            EnumNamingStrategy.SnakeCase => JoinWordsFrom("_", enumValueName),
+            EnumNamingStrategy.KebabCase => JoinWordsFrom("-", enumValueName),
+            _ => throw new ArgumentOutOfRangeException($"{nameof(namingStrategy)} has an unknown value {namingStrategy}"),
+        };
+    }
+
+    private static string[] SplitEnumValueNameIntoWords(string enumValueName)
+    {
+        // Matches (or):
+        // [A-Z][a-z]+
+        //  - A word starting with an uppercase letter, followed by lowercase letters ("Word" or "Word123")
+        // [A-Z]+(?![a-z])
+        //  - One or more uppercase letters, with no lowercase letters after ("ID" or "C1")
+        // \d+
+        //  - One or more digits ("123")
+        // "Pascal1CaseID2_3" = "Pascal1" + "Case" + "ID2" + "3"
+#pragma warning disable MA0009
+        var words = Regex.Matches(enumValueName, @"[A-Z][a-z]+|[A-Z]+(?![a-z])|\d+");
```

Comment:
I'd prefer a solution without regex a lot, shouldn't be too hard here and a lot I guess it should perform a lot better. Maybe use `char.GetUnicodeCategory`, `char.IsUpper`, `char.IsDigit`, ...

Reply:
Agreed, I have modified the solution to avoid using regex.